### PR TITLE
int64 support

### PR DESCRIPTION
**Pull request by ZanMaticPratnemer**
Support for int64 records and int64 type.

Additionally, rearanged the files into a current EPICS module structure.

### CODE REVIEW

**Comment by ZanMaticPratnemer** [comment]
We found another problem: CPU usage when inSize was 0.
Explained and fixed in: 199d793

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,5 @@ O.*
 bin
 lib
 dbd
+db
+*envPaths
```
Comment (dirk-zimoch) [inline .gitignore:7]: Fix missing newline!

File: GNUmakefile [hunk 2]
```diff
@@ -1,3 +1,5 @@
+## THIS FILE IS OUTDATED!!
+
```
Comment (dirk-zimoch) [inline GNUmakefile:2]: No, it isn't. It is PSI internal.

File: Makefile.Vx [hunk 3]
```diff
@@ -1,3 +1,6 @@
+## THIS FILE IS OUTDATED!!
```
Comment (dirk-zimoch) [inline Makefile.Vx:1]: It allows to build the module for EPICS 3.13. Which IS outdated but still in use at PSI.

File: S7plcApp/Db/example_int64.db [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  field (OUT, "@Testsystem0/96 T=INT64 B=63")
+  field (PINI, "YES")
+}
+
+record (bo, "bo-int64-1") {
+  field (DTYP, "S7plc")
+  field (DOL, "1")
+  field (OUT, "@Testsystem0/96 T=INT64 B=63")
+  field (PINI, "YES")
+}
```
Comment (dirk-zimoch) [inline S7plcApp/Db/example_int64.db:64]: Files without newline at the end will not be accepted.

**Comment by dirk-zimoch** [comment]
There are a few style guides which I follow and which need to be followed to be accepted for merge.
Please adjust your editor accordingly.
1. No text file ends without a newline.
2. No tabs for indents (except in Makefiles of course). Use 4 spaces.
3. No trailing whitespace at the end of a line.

**Comment by ZanMaticPratnemer** [comment]
Should I split this into 3 pull requests?
-file structure update,
-int64 support,
-CPU usage fix

**Comment by dirk-zimoch** [comment]
No need for that. Just give me some days for testing....

**Review by dirk-zimoch — CHANGES_REQUESTED** [review]
Please test compilation with older EPICS versions as well and avoid compiler warnings. I got warnings about unused variables with 3.14.12 and still a warning about a file with no newline at the end.

**Review by dirk-zimoch — APPROVED** [review]
Looks OK so far. I am now tesing with different EPICS base versions. Let's see how it works with 3.13.
I am not yet quite happy with the test for int64* record availability (>= 3.16.1).  Maybe simply checking if the headers are installed is easier. I will try...

**Comment by dirk-zimoch** [comment]
I changed the Makfile as follows:
```
# Check for int64 support
ifneq ($(wildcard $(EPICS_BASE)/include/int64*Record.h),)
EPICS_INT64 = true
endif
```
This checks if `in64*Record.h` files are available in base. That is much easier than the check against version number 3.16.1.

**Comment by dirk-zimoch** [comment]
There is an inconsitency in the products. We get a library `libs7plc.so` and an example executable `S7plcApp`. But there is only one dbd file `s7plc.dbd` and that is for the executable, not for the library. So the name does not even match.
There should be `s7plc.dbd` for `libs7plc.so` (no EPICS base stuff) and `S7plcApp.dbd` for `S7plcApp` (with full base stuff). Maybe capitalisation of "s7" should also be consistent between library and executable.

**Comment by dirk-zimoch** [comment]
I have now tested to build with 3.14.12, 3.16.0.1, 3.16.1, and 7.0.5.1. So far so good. But it does not build with 3.13.10 any longer. Maybe it is time to drop 3.13 support. In that case the code sould be cleaned up as well: All compiler branches on `EPICS_3_13` sould be removed as well as the 3.13 specific files `compat3_13.h` and `Makefile.Vx`.

**Comment by dirk-zimoch** [comment]
Woring on Windows compatibility now....

**Comment by dirk-zimoch** [comment]
Found a problem. I noticed that error are printet twice, because two threads (send and receive thread) are both calling s7plcConnect.
This was not the case before.

**Comment by dirk-zimoch** [comment]
There are several places in `ai` and `ao` record handling where hwHigh/hwLow are converted to 32 bit int, but they are potentially 64 bit now. Also the possibility that the raw value is 64 bit is not handled.

**Pull request merged**